cdap-ui/app/cdap/components/SecureKeys/SecureKeyDetails/index.tsx

```diff
@@ -0,0 +1,168 @@
+import Divider from '@material-ui/core/Divider';
+import FileCopyOutlinedIcon from '@material-ui/icons/FileCopyOutlined';
+import FormControl from '@material-ui/core/FormControl';
+import IconButton from '@material-ui/core/IconButton';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import Paper from '@material-ui/core/Paper';
+import TextField from '@material-ui/core/TextField';
+import classnames from 'classnames';
+import { copyToClipBoard } from 'services/Clipboard';
+
+const styles = (theme): StyleRules => {
+  return {
+    secureKeysTitle: {
+      paddingTop: theme.spacing(1),
+    },
+    divider: {
+      width: '100vw',
+    },
+    secureKeyDetails: {
+      margin: `${theme.Spacing(3)}px 0px`,
+      border: `1px solid ${theme.palette.grey[300]}`,
+      borderRadius: '6px',
+      padding: theme.spacing(3),
+      display: 'grid',
+      gridTemplateColumns: 'repeat(4, 1fr)',
+    },
+    keyID: {
+      gridColumnStart: '1',
+      gridColumnEnd: '2',
+    },
+    details: {
+      gridColumnStart: '2',
+      gridColumnEnd: '5',
+    },
+    secureKeyActionButton: {
+      margin: theme.spacing(1),
+      float: 'right',
+    },
+    secureKeyInput: {
+      margin: `${theme.Spacing(1)}px 0px`,
+      whiteSpace: 'nowrap',
+      wordWrap: 'normal',
+      paddingRight: '12px',
+      letterSpacing: '.00625em',
+      fontSize: '1rem',
+      lineHeight: '1.5rem',
+    },
+  };
+};
+
+interface ISecureKeysDetailsProps extends WithStyles<typeof styles> {
+  state: any;
+  onBackButtonClick: () => void;
+  openDeleteDialog: (index: number) => void;
+  openEditDialog: (index: number) => void;
+}
+
+const SecureKeysDetailsView: React.FC<ISecureKeysDetailsProps> = ({
+  classes,
+  state,
+  onBackButtonClick,
+  openDeleteDialog,
+  openEditDialog,
+}) => {
+  const { secureKeys, activeKeyIndex } = state;
+
+  const keyMetadata = secureKeys.get(activeKeyIndex);
+
+  return (
+    <div>
+      <h1 className={classes.secureKeysTitle}>
+        <IconButton onClick={onBackButtonClick}>
+          <ArrowBackIcon />
+        </IconButton>
+        {keyMetadata.get('name')}
+      </h1>
+      <Divider className={classes.divider} />
+      <Paper className={classes.secureKeyDetails} elevation={0}>
+        <div className={classes.keyID}>{keyMetadata.get('name')}</div>
+        <div className={classes.details}>
+          <FormControl
+            fullWidth
+            className={classnames(classes.margin, classes.textField)}
+            variant="outlined"
+          >
+            <TextField
+              type={'text'}
+              value={keyMetadata.get('description')}
+              variant="filled"
+              className={classes.secureKeyInput}
+              InputProps={{
+                readOnly: true,
+                endAdornment: (
+                  <InputAdornment position="end">
+                    <IconButton onClick={() => copyToClipBoard(keyMetadata.get('description'))}>
```

Comment:
I don't think you need a copyToClipboard for description of the secure key. I think if you want to add this feature, it's better for the name, because that is what is going to be used in pipelines

Reply:
temporarily deleted Details view, may add it after further discussion with lea.